### PR TITLE
chore: set data engine explicitly for the default storageclass

### PR DESCRIPTION
**Pull request by shuo-wu**
#### Which issue(s) this PR fixes:

https://github.com/longhorn/longhorn/issues/9584

### CODE REVIEW

**Comment by innobead** [comment]
> #### Which issue(s) this PR fixes:

@shuo-wu let's make sure an issue for this PR, so QA can help validate it.

**Comment by shuo-wu** [comment]
The ticket is created with a manual test attached.

**Comment by derekbit** [comment]
@shuo-wu 
Can you execute [generate-longhorn-yaml.sh](https://github.com/longhorn/longhorn/blob/master/scripts/generate-longhorn-yaml.sh)?

**Comment by shuo-wu** [comment]
@derekbit 

Done. Thank you for the reminder

File: chart/values.yaml [hunk 1]
```diff
@@ -172,6 +172,8 @@ persistence:
     selector: ""
   # -- Setting that allows you to enable automatic snapshot removal during filesystem trim for a Longhorn StorageClass. (Options: "ignored", "enabled", "disabled")
   removeSnapshotsDuringFilesystemTrim: ignored
+  # -- Setting that allows you to specify the data engine version for the default Longhorn StorageClass. (Options: "v1", "v2", "all")
```
Comment (innobead) [inline chart/values.yaml:175]: Just noticed we have `all` for data engine. What does that mean?

Comment (shuo-wu) [reply chart/values.yaml:175]: Just re-check the code. Option `all` is more like a special flag that [explicitly makes Longhorn list all 2 versions of instance managers/volumes](https://github.com/longhorn/longhorn-manager/blob/56903baa3890d0cac86d5a477f4863271b9043f9/types/types.go#L481-L483). But it seems that using an empty `dataEngine` could achieve the same result.

I am thinking if we should forbid users to use it.

Comment (shuo-wu) [reply chart/values.yaml:175]: OK. In Volume/Engine/Replica CRD, it allows `v1` and `v2` only. Let me remove option `all` for storageclass

Comment (innobead) [reply chart/values.yaml]: > Just re-check the code. Option `all` is more like a special flag that [explicitly makes Longhorn list all 2 versions of instance managers/volumes](https://github.com/longhorn/longhorn-manager/blob/56903baa3890d0cac86d5a477f4863271b9043f9/types/types.go#L481-L483). But it seems that using an empty `dataEngine` could achieve the same result.
> 
> I am thinking if we should forbid users to use it.

Yeah. I remembered that context discussed with @derekbit before. Just need a doc for that intended internal usage.

**Review by innobead — APPROVED** [review]

**Comment by shuo-wu** [comment]
@mergify backport to v1.7.x

**Pull request merged**